cmd/pay/main.go

```diff
@@ -95,6 +95,50 @@ var commands = map[string]cmd.Cmd{
 		Description:   "Interact with Temporal's various queue APIs",
 		ChildRequired: true,
 		Children: map[string]cmd.Cmd{
+			"ens": cmd.Cmd{
+				Blurb:       "ens queue command",
+				Description: "Used to launch ens request processing queue",
+				Action: func(cfg config.TemporalConfig, args map[string]string) {
+					logger, err := log.NewLogger(logPath(cfg.LogDir, "ens_consumer.log"), *devMode)
+					if err != nil {
+						fmt.Println("failed to start logger", err)
+						os.Exit(1)
+					}
+					db, err := newDB(cfg, *dbNoSSL)
+					if err != nil {
+						fmt.Println("failed to start db", err)
+						os.Exit(1)
+					}
+					quitChannel := make(chan os.Signal)
+					signal.Notify(quitChannel, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
+					waitGroup := &sync.WaitGroup{}
+					go func() {
+						fmt.Println(closeMessage)
+						<-quitChannel
+						cancel()
+					}()
+					for {
+						qm, err := queue.New(queue.ENSRequestQueue, &cfg, logger, *devMode)
+						if err != nil {
+							fmt.Println("failed to start queue", err)
+							os.Exit(1)
+						}
+						waitGroup.Add(1)
+						err = qm.ConsumeMessages(ctx, waitGroup, db, &cfg)
+						if err != nil && err.Error() != queue.ErrReconnect {
+							fmt.Println("failed to consume messages", err)
+							os.Exit(1)
+						} else if err != nil && err.Error() == queue.ErrReconnect {
+							continue
```

Comment:
I don't think this branch is useful. If err is not nil, for loop will continue naturally.

Reply:
good catch